Source/WebCore/Modules/reporting/ReportingScope.cpp

```diff
@@ -91,13 +92,18 @@ void ReportingScope::notifyReportObservers(Ref<Report>&& report)
     for (auto& observer : possibleReportObservers)
         observer->appendQueuedReportIfCorrectType(report);
 
+    auto currentReportType = report->body()->reportBodyType();
+
     // Step 4.2.2
+    m_queuedReportTypeCounts.add(currentReportType);
     m_queuedReports.append(WTFMove(report));
 
     // Step 4.2.3-4: If scope’s report buffer now contains more than 100 reports with type equal to type, remove the earliest item with type equal to type in the report buffer.
-    // FIXME(244369): When we support more than one time of report, remove only the specific type of report.
-    if (m_queuedReports.size() > 100)
-        m_queuedReports.removeFirst();
+    if (m_queuedReportTypeCounts.count(currentReportType) > 100) {
+        m_queuedReports.removeFirstMatching([currentReportType](auto& report) {
+            return report->body()->reportBodyType() == currentReportType;
+        });
```

Comment:
Don't we need to remove one from m_queuedReportTypeCounts too?

Reply:
Oh gosh, I must not have pushed my local change here. You are right -- this does need to be decremented, too.